runtime/runtime_events.c

```diff
@@ -376,6 +376,7 @@ static void runtime_events_create_from_stw_single(void) {
 
     // at the same instant: snapshot user_events list and set
     // runtime_events_enabled to 1
+    /* calling from STW */
```

Comment:
Note: I'm not sure I see the deadlock mentioned in this commit message. I thought about cases where a mutator would try to take the lock, but it is already held by a STW section. But the `runtime_events_create_from_stw_single` is invoked within a global barrier within the STW section, so the STW mechanism already excludes mutators from running at the same time. Where is the potential issue?

Reply:
One mutator waiting for the lock, another waiting for a STW while holding the lock.

The STW case above is a red herring (but the comment clarifies why we must use blocking rather than non_blocking).